### PR TITLE
DWS: handling for Lustre MGTs

### LINKED ISSUE #186: Rabbit DirectiveBreakdown AllocateSingleServer not working
Trying to create a standalone MGS through flux doesn't work. The "servers" resource is filled in with an empty "storage" field.

A standalone MGS can be created with "standaloneMgtPoolName" set to the pool name in the NnfStorageProfile. When a "#dw create_persistent type=lustre ..." workflow is created, a DirectiveBreakdown is made with a single allocation set asking for space for the MGT.

NnfStorageProfile:
```
...
data:
  default: false
  lustreStorage:
    capacityMgt: 5GiB
    combinedMgtMdt: false
    exclusiveMdt: false
    standaloneMgtPoolName: test-pool
...
```

DirectiveBreakdown:
```
...
status:
  ready: true
  storage:
    allocationSets:
    - allocationStrategy: AllocateSingleServer
      constraints:
        colocation:
        - key: lustre-mgt
          type: exclusive
        count: 1
        labels:
        - dataworkflowservices.github.io/storage=Rabbit
      label: mgt
      minimumCapacity: 5368709120
```

Flux fills in the "Servers" resource as follows:
```
    "spec": {
        "allocationSets": [
            {
                "allocationSize": 5368709120,
                "label": "mgt",
                "storage": []
            }
        ]
    },
```

From looking at the DWS code, it looks like the AllocateSingleServer case is not handled here: https://github.com/flux-framework/flux-coral2/blob/master/src/python/flux_k8s/directivebreakdown.py#L34 resulting in the empty "Storage" field.

I haven't tested it, but this would probably also cause a problem for a Lustre file system that allocated an MGT as part of the job (i.e., ExternalMgs not set).

### PR DESCRIPTION
**Pull request by jameshcorbett**
Problem: Flux does not handle workflows that require Lustre MGTs,
or more specifically any DirectiveBreakdown with an allocation type
of 'AllocateSingleServer'. However, Flux handles MDTs and other
allocations which have a 'count' constraint of 1, which is the same
thing effectively.

Handle 'AllocateSingleServer' allocations the same way that 'count'
constraints are handled (see #181 ), which is to pick a single rabbit and
place the allocation there, ignoring Fluxion's allocation choices
somewhat.

Fixes #186 .

### CODE REVIEW

File: t/t1002-dws-workflow-obj.t [hunk 1]
```diff
@@ -369,6 +369,22 @@ test_expect_success 'job submission with persistent DW string works' '
 	flux job wait-event -vt 30 ${jobid} clean
 '
 
+test_expect_success 'job submission with standalone MGT persistent DW string works' '
```
Comment (cmoussa1) [inline t/t1002-dws-workflow-obj.t:372]: Ignore this if I'm misunderstanding the acronyms - "MDT" here instead of "MGT"?

**Review by cmoussa1 — APPROVED** [review]
LGTM! One minor spelling comment.

File: t/t1002-dws-workflow-obj.t (hunk 1 above)
Comment (cmoussa1) [reply t/t1002-dws-workflow-obj.t:372]: Whoops, yeah you can probably just ignore this. Just realized "MGT" is in the commit title - my bad!!

Comment (jameshcorbett) [reply t/t1002-dws-workflow-obj.t:372]: No you're right, it looks like one commit has MDT and one has MGT. They should both be MGT. Good catch! I'll fix and set MWP.

**Pull request merged**